client/session_pools.go

```diff
@@ -57,7 +57,7 @@ func (s sessionPools) SeriesIterator() encoding.SeriesIteratorPool {
 	return s.seriesIterator
 }
 
-func (s sessionPools) IteratorArray() encoding.IteratorArrayPool {
+func (s sessionPools) IteratorArray() encoding.MultiReaderIteratorPool {
```

Comment:
as above, I don't think you need these changes. the pool you're after is already present in this struct/interface.

Reply:
Fixed as this is an array pool now